Sources/Algorithms/Trim.swift

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension BidirectionalCollection {
+
+  /// Returns a `SubSequence` formed by discarding all elements at the start and end of the Collection
+  /// which satisfy the given predicate.
+  ///
+  /// Trimming is a common operation in computing, often referring to removing leading and trailing whitespace
+  /// from a string[1].
+  ///
+  ///  ```
+  ///  let myString = "  hello, world  "
+  ///  print(myString.trimmed(where: \.isWhitespace)) // "hello, world"
+  ///  ```
+  ///
+  /// [1]: https://en.wikipedia.org/wiki/Trimming_(computer_programming)
```

Comment:
```suggestion
```

We should mention that if the entire collection is trimmed, the  result is an empty subsequence at the end of the collection.

Reply:
Personally I'd prefer to leave the position of the empty slice unspecified. If we decided we wanted to change this to first trim elements from the tail and then the head, the result when trimming the entire collection would be an empty subsequence at `startIndex`.